lib/compress.js

```diff
@@ -1731,6 +1731,7 @@ merge(Compressor.prototype, {
                 var val = {};
                 for (var i = 0, len = this.properties.length; i < len; i++) {
                     var prop = this.properties[i];
+                    if (prop instanceof AST_Expansion) return this;
```

Comment:
> That expression behaves the same with or without the patch.

In that case, my mistake then.

Given `({a:b})[c]`, if we want to find out whether it is side-effect-free, we do `may_throw_on_access` on `({a:b})` and see if it returns `false`.

> Would you be able to put together a fix for both array and object spreads on harmony?

Not immediately as I'm nowhere near a machine I can test things properly on. Sorry about that!